### DISCUSSION TITLE
How to create/preserve a frontmatter in Markdown/GFM output?

### DISCUSSION THREAD

**Discussion by mavam** [quarto-dev/quarto-cli#2276] [Q&A] [answered]
I'd like to know if I can include and/or preserve a YAML frontmatter in the Markdown and GFM output.

**Use case**: I'd like to use a Quarto notebook as blog post. The blogging platform takes GFM as input (Docusaurus). The blog post meta data is in a YAML frontmatter.

**Workaround**: concatenate the desired YAML frontmatter from an external file. This is suboptimal, as it breaks the self-containment and requires extra build scaffolding.

**Desired functionality**: provide a flag to `quarto build` that preserves the YAML frontmatter, or a subset thereof.

**Comment by mavam** [comment, marked as answer]
I should have taken a closer look at the docs. Via projects, adding this to `_quarto.yml` does the job:

```
format:
  gfm:
    keep-yaml: true
```

Any *.qmd files now keep the YAML frontmatter intact.

**Reply by ghost** [reply]
Thanks for the follow-up.
